src/jdk.incubator.code/share/classes/jdk/incubator/code/type/CoreTypeFactory.java

```diff
@@ -128,10 +128,14 @@ public JavaType constructType(TypeElement.ExternalizedTypeElement tree) {
                             (ClassType)constructType(parseExTypeElem(parts[0])),
                             constructTypeArgument(tree, 0, NO_WILDCARDS));
                 } else if (parts.length == 3) {
-                    // method type-var
-                    return JavaType.typeVarRef(parts[2],
-                            parseMethodRef(String.format("%s::%s", parts[0], parts[1])),
-                            constructTypeArgument(tree, 0, NO_WILDCARDS));
+                    // method or constructor type-var
```

Comment:
This strongly indicates we need to make refs be type elements, perhaps extending from say `JavaRef`, representing the common super class for symbolic references to Java things distinct from a `JavaType`. They formally become part of the code model and then we can manage this more uniformly. We can revisit later (including the distinction between structured string encodings and human readable encodings for Java types and refs).